src/Lean/Elab/Binders.lean

```diff
@@ -552,7 +558,7 @@ def expandMatchAltsWhereDecls (matchAltsWhereDecls : Syntax) : MacroM Syntax :=
 
 open Lean.Elab.Term.Quotation in
 @[builtinQuotPrecheck Lean.Parser.Term.fun] def precheckFun : Precheck
-  | `(fun $binders* => $body) => do
+  | `(fun $binders* $[: $ty?]? => $body) => do
```

Comment:
Should we use `_ty` to avoid the warning from the unused variable linter?

Reply:
Ah, not sure why I didn't see the warning. Will fix when I'm home.